Application/Setting/Univention/Service.php

```diff
@@ -0,0 +1,623 @@
+                            $SchoolTypeString = Type::useService()->getSchoolTypeString($tblSchoolType);
+                            $SchoolString = $this->getSchoolString($Acronym, $SchoolTypeString, $tblCompany);
+                            $schools[] = $SchoolString;
+                            $StudentSchool = $SchoolString;
+                        }
+                    }
+                }
+            }
+
+            if (!empty($schools)){
+                $schools = array_unique($schools);
+                $Item['schools'] = implode(',', $schools);
+            } else {
+                if(isset($TeacherSchools[$tblPerson->getId()])){
+                    $SchoolList = $TeacherSchools[$tblPerson->getId()];
+                    $SchoolList = array_unique($SchoolList);
+                    sort($SchoolList);
+                    $Item['schools'] = implode(',', $SchoolList);
+                }
+            }
+
+//            // Uploadtest
+//            //ToDO use API Schools
+//            if($tblStudent = Student::useService()->getStudentByPerson($tblPerson)){
+////                if(rand(0, 1)){
+//                $schools = array(
+////                        $schoolList['DEMOSCHOOL']
+//                    'DEMOSCHOOL'
+//                );
+//                $StudentSchool = 'DEMOSCHOOL';
+////                } else {
+////                    $schools = array(
+////                        $schoolList['DEMOSCHOOL2']
+////                    );
+////                }
+//            } else {
+//                $schools = array(
+//                    'DEMOSCHOOL',
+//                    'DEMOSCHOOL2'
+////                    $schoolList['DEMOSCHOOL'],
+////                    $schoolList['DEMOSCHOOL2']
+//                );
+//            }
+
+            if($tblDivision){
+                $UploadItem['school_classes'] = $StudentSchool.'-'.$tblDivision->getTblLevel()->getName().$tblDivision->getName();
+                $tblDivisionStudent = Division::useService()->getDivisionStudentByDivisionAndPerson($tblDivision, $tblPerson);
+            } else {
+                if(isset($TeacherClasses[$tblPerson->getId()])){
+                    $ClassList = $TeacherClasses[$tblPerson->getId()];
+                    foreach($ClassList as &$Class){
+                        $Class = $StudentSchool.'-'.$Class;
+                    }
+                    sort($ClassList);
+                    $Item['school_classes'] = implode(',', $ClassList);
+                }
+            }
+
+            if(!empty($schools)){
+                $schools = array_unique($schools);
+            }
+
+            $UploadItem['schools'] = $schools;
+
+            array_push($activeAccountList, $UploadItem);
+        });
+
+        return $activeAccountList;
+    }
+
+    /**
+     * @param bool $StudentWithoutAccount
+     *
+     * @return false|array
+     */
+    public function getExportAccount($StudentWithoutAccount = true)
```

Comment:
Codacy found an issue: [The method getExportAccount() has 102 lines of code. Current threshold is set to 100. Avoid really long methods.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)